integration-tests/contract-security-tickets-join.spec.ts

```diff
@@ -0,0 +1,229 @@
+
+// TC-T-018: Create ticket - call input string/address
+// TC-T-019: Create ticket -  string/address
+// TC-T-020: Join ticket - two different ticketers
+// TC-T-021: Join ticket - two different, but similar cty
+
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib;
+
+  describe(`Test contracts using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    it('Verify creating ticket is not possible with call input string/address', async (done) => {
+      try {
+        const opContract = await Tezos.contract.originate({
+          code: `   {  parameter (ticket string);
+        storage (option (ticket string));
+        code
+          {
+            CAR; # drop storage, but leave input on stack
+            PUSH nat 4;
+            PUSH string "test";
+            TICKET;
+            PAIR;
+            JOIN_TICKETS;
+            NIL operation;
+            PAIR;
+          };}`,
+          init: 'Unit',
+        });
+
+        await opContract.confirmation();
+        expect(opContract.hash).toBeDefined();
+        expect(opContract.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opContractContract = await opContract.contract();
+        // expect(await opContractContract.storage()).toBeTruthy();
+
+        await Tezos.contract
+          .at(opContractContract.address)
+          .then((contract) => {
+            return contract.methods
+              .default(`(Pair ${opContractContract.address} \"test\" 1)`)
+              .send();
+          })
+          .then((op) => {
+            return op.confirmation().then(() => op.hash);
+          });
+      } catch (error: any) {
+        expect(error.message).toContain('michelson_v1.invalid_primitive');
+      }
+      done();
+    });
+
+    it('Verify creating ticket is not possible with string/address', async (done) => {
+      try {
+        const opContract = await Tezos.contract.originate({
+          code: `   {  parameter (option (ticket string));
+        storage (option (ticket string));
+        code
+          {
+            CAR; # drop storage, but leave input on stack
+            IF_NONE { FAIL} {};
+            PUSH nat 4;
+            PUSH string "test";
+            TICKET;
+            PAIR;
+            JOIN_TICKETS;
+            NIL operation;
+            PAIR;
+          };
+        }`,
+          init: 'None',
+        });
+
+        await opContract.confirmation();
+        expect(opContract.hash).toBeDefined();
+        expect(opContract.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opContractContract = await opContract.contract();
+        // expect(await opContractContract.storage()).toBeTruthy();
+
+        await Tezos.contract
+          .at(opContractContract.address)
+          .then((contract) => {
+            return contract.methods
+              .default(`Some (Pair ${opContractContract.address} \"test\" 1)`)
```

Comment:
smart contract method can not be called with michelson in taquito

Reply:
dropped the Some